src/validationinterface.h

```diff
@@ -131,9 +132,18 @@ class CValidationInterface {
      * Called on a background thread.
      */
     virtual void TransactionRemovedFromMempool(const CTransactionRef& tx, MemPoolRemovalReason reason, uint64_t mempool_sequence) {}
+    /* Notifies listeners of transactions removed from the mempool as
+     * as a result of new block being connected.
+     * MempoolTransactionsRemovedForConnectedBlock will be fired before BlockConnected.
+     *
+     * Called on a background thread.
+     */
+    virtual void MempoolTransactionsRemovedForConnectedBlock(const std::vector<NewMempoolTransactionInfo>& txs_removed_for_block, unsigned int nBlockHeight) {}
```

Comment:
Thank you  @glozow
 Now using two structs `RemovedMempoolTransactionInfo` for this callback and `NewMempoolTransactionInfo` for `TransactionAddedToMempool callback.

Since they have similar fields, I created a base struct with all the similar fields so that they can both inherit 79803d3912747eb96153357f1cc33c3f8cf21eb6 and a469d937dbbadedeb58e0d5d1c44ed7881a7ebb9